lib/miq_expression.rb

```diff
@@ -1343,6 +1352,11 @@ def to_arel(exp, tz)
       escape = nil
       case_sensitive = true
       arel_attribute.matches("%#{parsed_value}%", escape, case_sensitive)
+    when "includes all", "includes any", "includes only"
```

Comment:
I wonder if this will be called if the sql query is partially sql friendly.
Can we add a `respond_to?` check here as well?

Reply:
I don't think this is necessary:

The only call to `to_arel` is [in `MiqExpression#to_sql`](https://github.com/NickLaMuro/manageiq/blob/140ec594/lib/miq_expression.rb#L289):

https://github.com/search?q=org%3AManageIQ+to_arel&type=Code

And in the line previous, we [call `#preprocess_for_sql`](https://github.com/NickLaMuro/manageiq/blob/140ec594/lib/miq_expression.rb#L294), which will make the check in for the change I added [in `#sql_supports_atom?`](https://github.com/NickLaMuro/manageiq/blob/140ec594/lib/miq_expression.rb#L347), so I think we don't need an additional check here.

Reply:
Also, `#preprocess_for_sql` will remove any non-SQL-friendly portions of the expression, so accidentally calling this when it doesn't support it will not happen.

Reply:
Is there value in "including" `includes all` and `includes only` if we're not going to get here?

Reply:
I think this is answered in https://github.com/ManageIQ/manageiq/pull/17562/files#r194541491

Reply:
> (and yes... I noticed the pun...)

Yay!